File: web_login_flutter/lib/models/form/choice.dart
import 'package:equatable/equatable.dart';
import 'package:web_login_flutter/models/form/question.dart';

class Choice extends Equatable {
  final int _id;
  final String _name;
  final Question? _nextQuestion;

  const Choice({id, name, nextQuestion}):
      _id = id, _name = name, _nextQuestion = nextQuestion;

  @override
  List<Object?> get props => [_id, _name, _nextQuestion];

  int get id => _id;

  Question? get nextQuestion => _nextQuestion;

  String get name => _name;

  @override
  String toString() {
    return 'Choice{_id: $_id, _name: $_name, _nextQuestion: $_nextQuestion}';
  }
}

File: web-login-core/src/main/java/com/example/weblogincore/domain/model/form/questions/Choice.java
package com.example.weblogincore.domain.model.form.questions;

public class Choice {
    private final Long id;
    private final String name;
    private final Question nextQuestion;

    public Choice(Long id, String name, Question nextQuestion) {
        this.id = id;
        this.name = name;
        this.nextQuestion = nextQuestion;
    }

    public String getName() {
        return name;
    }

    public Question getNextQuestion() {
        return nextQuestion;
    }

    public Long getId() {
        return id;
    }
}
